jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js

```diff
@@ -5,16 +6,94 @@ XXX jcssackett 2015-09-18: Licensing for juju.js? It's different then the
 licensing for the GUI.
 */
 
-;
+'use strict';
+
+var module = module;
+
+/**
+ * jujulib provides api access for microservices used by juju.
+ *
+ * jujulib provies access to the APIs for the Juju Environment
+ * Manager (JEM), the juju charmstore, and the juju identity
+ * manager (IdM).
+ */
+(function (exports) {
+
+    /**
+     * Environment object for jujulib.
+     *
+     * Provides access to the JEM API.
+     */
 
-var jujulib = (function (exports) {
+    /**
+     * Initializer
+     *
+     * @function environment
+     * @param url {String} The url, including scheme and port, of the JEM instance.
+     * @param bakery {Object} A bakery object for communicating with the JEM instance.
+     * @returns {Object}
+     */
+    var environment = function(url, bakery) {
```

Comment:
If I am understanding this correctly is this to be the constructor? A more common pattern is as follows:

```JavaScript
function environment(url, bakery) {
  this.bakery = bakery;
  this.jemUrl = url + '/v1';
}

environment.prototype._makeRequest = function() { ... }
environment.prototype.listEnvironments = function() { ... }
environment.prototype.getEnvironment = function() { ... }

```